extensions/amp-inline-gallery/1.0/thumbnails.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as Preact from '../../../src/preact';
+import {BaseCarousel} from '../../amp-base-carousel/1.0/base-carousel';
+import {CarouselContext} from '../../amp-base-carousel/1.0/carousel-context';
+import {cloneElement} from '../../../src/preact';
+import {px} from '../../../src/style';
+
+const DEFAULT_HEIGHT = 100;
+
+/**
+ * @param {!InlineGalleryDef.ThumbnailProps} props
+ * @return {PreactDef.Renderable}
+ */
+export function Thumbnails({
+  aspectRatio,
+  children,
+  loop = false,
+  style,
+  ...rest
+}) {
+  const pointerFine = window.matchMedia('(pointer: fine)');
+  const slideHeight = (style && style.height) || DEFAULT_HEIGHT;
+  // Note: The carousel is aria-hidden since it just duplicates the
+  // information of the original carousel.
+  return (
+    <CarouselContext.Consumer>
```

Comment:
Why not `useContext`?

Reply:
No particular reason, I can use that instead if preferred.

Reply:
Yeah. I think it ends up a bit cleaner.